Library/Homebrew/github_runner_matrix.rb

```diff
@@ -129,6 +129,23 @@ def runner_enabled?(macos_version)
     macos_version <= NEWEST_HOMEBREW_CORE_MACOS_RUNNER && macos_version >= OLDEST_HOMEBREW_CORE_MACOS_RUNNER
   end
 
+  NEW_INTEL_MACOS_MUST_BUILD_FORMULAE = %w[pkg-config pkgconf].freeze
+
+  sig { returns(T::Boolean) }
+  def deploy_new_x86_64_runner?
+    return true if @testing_formulae.any? { |f| NEW_INTEL_MACOS_MUST_BUILD_FORMULAE.include?(f.name) }
+    return true if @testing_formulae.any? { |f| f.formula.class.pour_bottle_only_if == :clt_installed }
+
+    Formula.all.any? do |formula|
+      non_test_dependencies = Dependency.expand(formula, cache_key: "determine-test-runners") do |_, dependency|
+        Dependency.prune if dependency.test?
+      end
+      next false if non_test_dependencies.none? { |dep| @testing_formulae.map(&:name).include?(dep.name) }
+
+      formula.class.pour_bottle_only_if == :clt_installed
```

Comment:
I'd move this before the dependency check (`return false if formula.class.pour_bottle_only_if != :clt_installed`). This should make it significantly faster.

Reply:
Are you referring to

https://github.com/Homebrew/brew/blob/41c5f7243f6813b17d8f05c2f862c5869049d04d/Library/Homebrew/github_runner_matrix.rb#L137

? Or am I misunderstanding you?

Reply:
No, within the `Formula.all?` block.

Reply:
Added a suggestion diff though might be hard to read.